### PR TITLE
chpldoc Spawn.chpl

### PR DESCRIPTION
**Pull request by mppf**
Add chpldoc documentation for the Spawn.chpl module

### CODE REVIEW

**Comment by mppf** [comment]
@lydia-duncan could you look at this one, too?

File: modules/standard/Spawn.chpl [hunk 1]
```diff
@@ -19,6 +19,69 @@
 
 
 /*
+
+Support launching and interacting with other programs.
+
+Using functions in this module, one can create a subprocess
+possibly and capture its output. It is also possible to provide
```
Comment (lydia-duncan) [inline modules/standard/Spawn.chpl]: Should "possibly" be here?

File: modules/standard/Spawn.chpl [hunk 2]
```diff
@@ -45,41 +108,74 @@ module Spawn {
...
+       for any channels that are necessary. */
     param kind:iokind;
+    /* As with kind, this value is used to create the types
+       for any channels that are necessary. */
     param locking:bool;
 
+    pragma "no doc"
     var home:locale;
 
+    /* The Process ID number of the spawned process */
```
Comment (lydia-duncan) [inline modules/standard/Spawn.chpl]: Process ID, and the file descriptors seems a little too much of a C concept for me.  Should we be exposing them?

File: modules/standard/Spawn.chpl [hunk 3]
```diff
@@ -266,6 +433,62 @@ module Spawn {
...
+                     
+     :arg shellarg: An argument to pass to the shell before
+                    the command string. By default this is "-c".
+
+     :arg kind: What kind of channels should be created when
+                :const:`PIPE` is used. This argument is used to set
+                :attr:`subprocess.kind` in the resulting subprocess.
+                Defaults to :enum:`IO.iokind.dynamic`.
+                 
+     :arg kind: Should channels created use locking?
```
Comment (lydia-duncan) [inline modules/standard/Spawn.chpl]: Should be "locking" instead of "kind"

File: modules/standard/Spawn.chpl (hunk 2 above)
Comment (mppf) [inline modules/standard/Spawn.chpl]: We could no-doc these things, that would be fine with me.
They aren't C concepts, they are operating systems (linux/POSIX/whatever) concepts.
Part of the purpose of this module is to interact with the operating system, so I think
it makes some sense that they can be exposed.

That said, I could explain them better or refer the reader somewhere else for details.

File: modules/standard/Spawn.chpl [hunk 4]
```diff
@@ -277,6 +500,11 @@ module Spawn {
     return spawn(args, env, executable, kind=kind, locking=locking);
   }
 
+  /*
+     Check to see if a child process has terminated.
+     If the child process has terminated, after this
+     call, :attr:`subprocess.running` will be `false`.
```
Comment (lydia-duncan) [inline modules/standard/Spawn.chpl]: Maybe truncate here?

File: modules/standard/Spawn.chpl [hunk 5]
```diff
@@ -289,6 +517,27 @@ module Spawn {
...
+  /*
+    Wait for a child process to complete. After this function
+    returns, :attr:`subprocess.running` is `false` and
+    :attr:`subprocess.exit_status` stores the exit code returned
+    by the subprocess.
+     
+    .. note::
+     
+        Use :proc:`subprocess.communicate` instead of this function when using
+        PIPE for stdin, stdout, or stderr.  This function does not try to send
```
Comment (lydia-duncan) [inline modules/standard/Spawn.chpl]: Do you want PIPE to be a reference here?

File: modules/standard/Spawn.chpl (hunk 2 above)
Comment (lydia-duncan) [inline modules/standard/Spawn.chpl]: Yeah, by "C concept", I really meant "too low level".

Comment (lydia-duncan) [inline modules/standard/Spawn.chpl]: I can see leaving PID in, but I'd be less okay with the file descriptors being exposed.

**Comment by lydia-duncan** [comment]
Okay, I'm done looking through!

File: modules/standard/Spawn.chpl (hunk 2 above)
Comment (mppf) [inline modules/standard/Spawn.chpl]: OK, I left in PID but no-doc'd the inputfd etc. It's harder to imagine a situation in which somebody wants to access those parts of the subprocess record.

Comment (lydia-duncan) [inline modules/standard/Spawn.chpl]: Sounds good, thanks!

**Comment by lydia-duncan** [comment]
I noticed when reading through the chpldoc .rst file that one could use a "versionadded" directive to list when the feature was added to the library or API.  Would you like to use that for this module?

**Comment by mppf** [comment]
I did that in #2601. I'd like to see version added at the end of the module block, personally, but that is not what our readme says to do. It could be more nicely formatted.

**Pull request merged**